core/src/stored/dir_cmd.cc

```diff
@@ -455,6 +459,50 @@ static bool SetdebugCmd(JobControlRecord* jcr)
   }
 }
 
+static DeviceResource* GetDeviceResource(const std::string& name)
+{
+  return static_cast<DeviceResource*>(
+      my_config->GetResWithName(R_DEVICE, name.c_str()));
+}
+
+static bool SetdeviceCmd(JobControlRecord* jcr)
+{
+  BareosSocket* dir = jcr->dir_bsock;
+
+  Dmsg1(10, "SetdeviceCmd: %s", dir->msg);
+
+  std::vector<char> device_name(128);  // trailing zero included
+  std::vector<char> autoselect_value(128);
+  int scan = sscanf(dir->msg, setdevice_autoselect, device_name.data(),
+                    autoselect_value.data());
+  if (scan != 2) {
+    dir->fsend(BADcmd, "setdevice", dir->msg);
+    return false;
+  }
+
+  auto res = GetDeviceResource(device_name.data());
+  if (res == nullptr) {
+    std::string err{"Device not found: "};
+    err += device_name.data();
+    dir->fsend(BADcmd, "setdevice", err.c_str());
+    return false;
+  }
+
+  if (autoselect_value.data() == std::string("on")) {
```

Comment:
Maybe we should also accept "1" for true and "0" for false as this is what is being printed in the status output.
Accepting yes and no probably also would be userfriendly.

Reply:
as a user I'd assume that I can use every syntax that is allowed for a boolean value in the configuration file